backend/corpora/common/upload.py

```diff
@@ -64,44 +72,39 @@ def upload(
         raise InvalidFileFormatException(f"{url} must be in the file format(s): {allowed_file_formats}")
 
     # Check if datasets can be added to the collection
-    collection = Collection.get_collection(
-        db_session,
-        collection_id,
-        visibility=CollectionVisibility.PRIVATE,  # Do not allow changes to public Collections
-        owner=owner_or_allowed(user, scope) if scope else user,
-    )
+    collection = db.check_collection_access(collection_id)
     if not collection:
         raise NonExistentCollectionException(f"Collection {collection_id} does not exist")
 
     # Check if a dataset already exists
     if dataset_id:
-        dataset = Dataset.get(db_session, dataset_id, collection_id=collection_id)
+        dataset = db.get_dataset(dataset_id)
         if not dataset:
             raise NonExistentDatasetException(f"Dataset {dataset_id} does not exist")
-    elif curator_tag:
-        dataset = Dataset.get_dataset_from_curator_tag(db_session, collection_id, curator_tag)
+    # elif curator_tag: # TODO: Curator API support
+    #     dataset = Dataset.get_dataset_from_curator_tag(db_session, collection_id, curator_tag)
     else:
         dataset = None
 
     if dataset:
         # Update dataset
-        if dataset.processing_status.processing_status not in [
+        if dataset['processing_status']['processing_status'] not in [
             ProcessingStatus.SUCCESS,
             ProcessingStatus.FAILURE,
         ]:
             raise InvalidProcessingStateException(
-                f"Unable to reprocess dataset {dataset_id}: {dataset.processing_status.processing_status=}"
+                f"Unable to reprocess dataset {dataset_id}: {dataset['processing_status']['processing_status']}"
             )
         else:
-            dataset.reprocess()
+            db.bulk_edit_dataset(dataset_id, Utils.empty_dataset)
 
     else:
         # Add new dataset
-        dataset = Dataset.create(db_session, collection=collection, curator_tag=curator_tag)
+        dataset_id = db.add_dataset(collection_id, Utils.empty_dataset)
 
-    dataset.update(processing_status=dataset.new_processing_status())
+    db.edit_dataset(dataset_id, "processing_status", new_processing_status())
```

Comment:
Mostly a nitpick, but I would keep the methods that change the dataset status separate from `edit_dataset`. In fact, it would probably be a good idea to add a check to `edit_dataset` so that only a dataset that belongs to a private collection can be edited.